src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1415,12 +1415,12 @@ class MacroAssembler: public Assembler {
 
   public:
 
-  void ldr_constant(Register dest, const Address &const_addr) {
-    if (NearCpool) {
+  void ldr_patchable(Register dest, const Address &const_addr) {
+    if (CodeCache::contains(const_addr.target())) {
```

Comment:
Doesn't this cause us to generate LDR for oops outside the code cache, when what we need is the ADRP below?  The caller is still using a PC-relative dummy address.

Reply:
(written above) in ldr_patchable adrp+ldr path is used.

Code generator encodes ldr to dummy address (real address is stored in oop_Relocation). During relocation, adrp+ldr are patched to a real address. This code only works for UseShenandoahGC and ldr_patchable only uses the adrp+ldr path.